### PR TITLE
Fw 4673 hide the media search boxes and switch over media ap is

### PR DESCRIPTION
**Pull request by sahil97**
### Description of Changes
- Updated media related components to hook up with the new APIs and new doc types
- Hid search box when adding media

### Checklist

- ~README / documentation has been updated if needed~
- [x] PR title / commit messages contain Jira ticket number if applicable
- ~Tests have been updated / created if applicable~

### Reviewers Should Do The Following:

- [x] Review the changes here
- ~Pull the branch and test locally~

### Additional Notes

N/A

### CODE REVIEW

File: src/common/utils/mediaHelpers.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ export const isDisplayablePropMedia = (property, value) =>
   (typeof value === 'string' || value instanceof String) &&
   DISPLAYABLE_PROPS_MEDIA.includes(property)
 
-export const selectOneFormHelper = (formData, mediaObjectkey) => {
+export const selectOneMediaFormHelper = (formData, mediaObjectkey) => {
```
Comment (sahil97) [inline src/common/utils/mediaHelpers.js:43]: Changed selectOne to selectOneMedia alongside all the helper functions.

File: src/components/ErrorHandler/ErrorHandlerContainer.js [hunk 2]
```diff
@@ -8,7 +8,7 @@ function ErrorHandlerContainer({ error }) {
   let errorStatus = null
   let errorText = null
 
-  if (error && error?.status) {
+  if (error?.status) {
```
Comment (sahil97) [inline src/components/ErrorHandler/ErrorHandlerContainer.js:11]: sonar wanted it concise

File: src/components/Form/ArrayBrowserField.js [hunk 3]
```diff
@@ -24,22 +27,38 @@ function ArrayBrowserField({
...
+                    type={docType}
+                    size={SMALL}
+                    imageStyles="object-cover pointer-events-none"
+                  />
+                )}
+                {docType === VIDEO && (
+                  <MediaThumbnail.Video id={docId} type={docType} />
+                )}
+                {/* For all other types of documents. */}
+                {![AUDIO, IMAGE, VIDEO].includes(docType) && (
```
Comment (sahil97) [inline src/components/Form/ArrayBrowserField.js:50]: Still kept the DocumentThumbnail till we move all documents to new data types and completely do away with the usage of getDocument endpoint.

File: src/components/Form/RadioButtons.js [hunk 4]
```diff
@@ -4,7 +4,7 @@ import { RadioGroup } from '@headlessui/react'
 import { Controller } from 'react-hook-form'
 
 // FPCC
-import Form from 'components/Form'
+import ErrorMessage from 'components/Form/ErrorMessage'
```
Comment (sahil97) [inline src/components/Form/RadioButtons.js:7]: This along with other similar imports in forms/or related to forms were refactored to resolve cyclic dependency issues.

File: src/components/Form/SelectOneMedia.js [hunk 5]
```diff
@@ -73,10 +73,18 @@ function SelectOneButton({ value, onChange }) {
...
     setAddMediaModalOpen(true)
   }
-
   return value?.docId ? (
     <div className="mt-1 inline-flex border border-transparent bg-white rounded-lg shadow-md text-sm font-medium p-2 space-x-1">
-      <DocumentThumbnail.Container docId={value?.docId} />
+      {docType === IMAGE ? (
+        <MediaThumbnail.Image
+          id={value?.docId}
+          type={docType}
```
Comment (gmcauliffe) [inline src/components/Form/SelectOneMedia.js:81]: The type here will always be 'image', and likely the size will always be small, so we could set them inside the `MediaThumbnail.Image` component and avoid passing extra props.

Same for `MediaThumbnail.Video`.

**Review by gmcauliffe — APPROVED** [review]
Thank you for your fast work on this 🥇

File: src/components/Form/SelectOneMedia.js (hunk 5 above)
Comment (sahil97) [reply src/components/Form/SelectOneMedia.js:81]: Yes, I missed that, good catch.

**Pull request merged**